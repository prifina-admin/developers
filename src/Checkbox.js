import React, {Component, forwardRef} from 'react';
import styled from 'styled-components';
import PropTypes from "prop-types";
import {default as styledProps} from "@styled-system/prop-types";
import {
    border,
    layout,
    space,
    compose,
    color,
    background,
    borderColor,
    borderRadius,
} from "styled-system";

const systemProps = compose(
    layout,
    space,
    border,
    color,
    background,
    borderColor,
    borderRadius,
);

const CheckboxElement = styled.input`
    ${systemProps};
`;

const Checkbox =  forwardRef((props, ref) => {
    return <CheckboxElement type="checkbox" {...props} ref={ref} />;
});

Checkbox.propTypes = {
    ...styledProps.space,
    ...styledProps.background,
    ...styledProps.color,
    ...styledProps.layout,
    ...styledProps.border,
    ...styledProps.borderColor,
    ...styledProps.borderRadius,
}

export default Checkbox;