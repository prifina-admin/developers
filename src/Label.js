import React, { forwardRef } from "react";

import styled, { css } from "styled-components";

import { layout, space, typography, color, compose } from "styled-system";
import { default as styledProps } from "@styled-system/prop-types";
import PropTypes from "prop-types";

const systemProps = compose(layout, space, typography, color);

const nowrap = props => {
  return props.nowrap
    ? {
        whiteSpace: "nowrap",
      }
    : null;
};
const pointer = props => {
  return props.pointer ? { cursor: "pointer" } : null;
};
const hidden = props => {
  return props.hidden ? { visibility: "hidden" } : null;
};

/*
const VisuallyHidden = styled(Box)`
  border: 0px;
  clip: rect(0px, 0px, 0px, 0px);
  height: 1px;
  width: 1px;
  margin: -1px;
  padding: 0px;
  overflow: hidden;
  white-space: nowrap;
  position: absolute;
`;
*/

const LabelTheme = css`
  font-size: ${props => props.theme.textStyles[props.textStyle].fontSize};
  letter-spacing: ${props =>
    props.theme.textStyles[props.textStyle].letterSpacing};
  display: ${props => props.variation};
  width: 100%;
  margin: 0;
  color: ${props => props.theme.textStyles[props.textStyle].color};
  font-weight: ${props => props.theme.textStyles[props.textStyle].fontWeight};
  line-height: ${props => props.theme.textStyles[props.textStyle].lineHeight};
  ${systemProps};
  ${nowrap}
  ${pointer}
  ${hidden}
`;

const LabelElement = styled("label")`
  ${LabelTheme}
`;

const Label = forwardRef((props, ref) => {
  return <LabelElement {...props} ref={ref} />;
});
Label.propTypes = {
  ...styledProps.space,
  ...styledProps.layout,
  ...styledProps.typography,

  variation: PropTypes.oneOf(["block", "inline", "none"]),
  pointer: PropTypes.bool,
};
Label.defaultProps = {
  textStyle: "body",
};

Label.displayName = "Label";
Label.isLabel = true;

export default Label;
