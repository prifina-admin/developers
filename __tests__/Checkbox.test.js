import React, {createRef} from "react";
import renderer from "react-test-renderer";
import Checkbox from "../src/Checkbox";
import Label from "../src/Label";
import styled from "styled-components";
import {action} from "@storybook/addon-actions";
import ThemeProvider from "../src/theme/ThemeProvider";

const StyledLabel = styled(Label)`
  cursor: pointer;
  vertical-align: middle;
`;

const checkAction = e => {
    action(`${e.target.id} was clicked`)(e.target.value, e.target.checked);
};

test("Checkbox renders correctly", () => {
    const tree = renderer.create(<ThemeProvider>
        <StyledLabel htmlFor="unchecked_box">
            <Checkbox id="unchecked_box" onChange={checkAction} />
            Unchecked by default
        </StyledLabel>
    </ThemeProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});

