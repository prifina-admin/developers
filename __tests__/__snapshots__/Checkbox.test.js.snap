// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox renders correctly 1`] = `
<div
  className="ThemeProvider__Base-sc-1rk52n4-0 emGfPA"
>
  <label
    className="Label__LabelElement-u5wfo1-0 cOOeVN Checkboxtest__StyledLabel-vd4n10-0 cctMno"
    htmlFor="unchecked_box"
  >
    <div
      className="Box__BoxElement-sc-1iw4lg9-0 dnpTCS Checkbox__CheckBoxWrapper-u1j579-1 zSkhQ"
    >
      <input
        className="Checkbox__StyledInput-u1j579-0 cibfPi"
        id="unchecked_box"
        onChange={[Function]}
        role="checkbox"
        type="checkbox"
      />
      <svg
        className="Icon-y3r77h-0 hfhfoT"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M19 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2m0 2v14H5V5h14m-9 12l-4-4l1.41-1.42L10 14.17l6.59-6.59L18 9\\" fill=\\"currentColor\\"/>",
          }
        }
        data-name="checked"
        focusable={false}
        height="1.5rem"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "transform": "rotate(360deg)",
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 24 24"
        width="1.5rem"
        xmlns="http://www.w3.org/2000/svg"
      />
      <svg
        className="Icon-y3r77h-0 hfhfoT"
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M19 3H5c-1.11 0-2 .89-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2V5a2 2 0 0 0-2-2m0 2v14H5V5h14z\\" fill=\\"currentColor\\"/>",
          }
        }
        data-name="empty"
        focusable={false}
        height="1.5rem"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "transform": "rotate(360deg)",
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 24 24"
        width="1.5rem"
        xmlns="http://www.w3.org/2000/svg"
      />
    </div>
    Unchecked by default
  </label>
</div>
`;
